app/grandchallenge/publications/utils/__init__.py

```diff
@@ -1,11 +1,75 @@
+import re
+
 import requests
+from django.core.validators import RegexValidator
+from django.db import models
+from django.forms import CharField
 
-from grandchallenge.publications.models import (
-    PublicationType,
-    get_publication_type,
-)
 from grandchallenge.publications.utils.manubot import get_arxiv_csl
 
+# regex modified for python syntax from
+# https://www.crossref.org/blog/dois-and-matching-regular-expressions/
+DOI_REGEX = r"^10\.\d{4,9}/[-._;()/:a-z0-9]+$"
+ARXIV_REGEX = r"^\d{4}\.\d{4,5}$"
+
+identifier_validator = RegexValidator(regex=f"{DOI_REGEX}|{ARXIV_REGEX}")
+
+
+class PublicationType(models.TextChoices):
+    DOI = "D"
+    ARXIV = "A"
+
+
+class IdentifierField(models.CharField):
+
+    description = (
+        "The DOI e.g., 10.1002/mrm.25227, or the arXiv id, e.g., 2006.12449"
+    )
+
+    def __init__(self, *args, **kwargs):
+        kwargs["max_length"] = 255
+        kwargs[
+            "help_text"
+        ] = "The DOI, e.g., 10.1002/mrm.25227, or the arXiv id, e.g., 2006.12449"
```

Comment:
Duplicate?

Reply:
The description is different from the help text. It's a way to document what the field is used for. It would be shown in the admin docs, but I just realized that we don't use that feature, so I guess I can leave the description out entirely.